nemo/src/io/resource_providers/http.rs

```diff
@@ -97,11 +145,14 @@ impl ResourceProvider for HttpResourceProvider {
         resource: &Resource,
         media_type: &str,
     ) -> Result<Option<Box<dyn Read>>, ReadingError> {
-        if !is_iri(resource) {
-            return Ok(None);
-        }
-
-        if !(resource.starts_with("http:") || resource.starts_with("https:")) {
+        // Add error message
+        let Resource::Iri { iri, parameters } = resource else {
+            unreachable!("The type of the resource is known already")
+        };
```

Comment:
This needs to go back to what it was before, to support proper selection of Resource providers.